executors/cpp/plural_rules.cpp

```diff
@@ -0,0 +1,171 @@
+/********************************************************************
+ * Comments and license as needed
+ ************************************
+
+/******
+ * testing plural rules
+ * Based on code of ICU4C testing:
+ * https://github.com/unicode-org/icu/blob/maint/maint-75/icu4c/source/test/intltest/plurfmts.cpp
+ */
+
+
+#include "unicode/utypes.h"
+#include "unicode/unistr.h"
+#include "unicode/plurfmt.h"
+#include "unicode/plurrule.h"
+
+#include "unicode/locid.h"
+#include "unicode/uclean.h"
+
+#include "util.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+
+#include <json-c/json.h>
+
+#include <iostream>
+#include <string>
+#include <string_view>
+#include <cstring>
+
+using icu::PluralRules;
+
+using std::cout;
+using std::endl;
+using std::string;
+
+
+const string test_plural_rules (json_object* json_in) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  json_object* label_obj = json_object_object_get(json_in, "label");
+  string label_string = json_object_get_string(label_obj);
+
+  // The locale in which the name is given.
+  string locale_string = "und";
+  string type_string = "";
+
+  json_object* locale_label_obj = json_object_object_get(json_in, "locale");
+  if (locale_label_obj) {
+    locale_string = json_object_get_string(locale_label_obj);
+  }
+  Locale display_locale(locale_string.c_str());
+
+  json_object* return_json = json_object_new_object();
+  json_object_object_add(return_json, "label", label_obj);
+
+  // The language's name to be displayed.
+  json_object* sample_obj = json_object_object_get(json_in, "sample");
+
+  bool input_is_double = false;
+  bool input_is_integer = false;
+  bool input_is_compact = false;
+
+  double input_double_sample;
+  int32_t input_int_sample;
+
+  std::vector<UnicodeString> u_strings;
+  int u_strings_size = 0;
+  if (sample_obj) {
+    // Get the number
+    string sample_string = json_object_get_string(sample_obj);
+
+    if (sample_string.find('.') != std::string::npos) {
+      // Convert into an integer, decimal, or compact decimal
+      input_double_sample = std::stod(sample_string);
+      input_is_double = true;
+      cout << "#    DOUBLE STRING " << input_double_sample << endl;
```

Comment:
Remove low level stdout printing statements, or else use a logging library and convert to debug level log line.

update: based on the TODO statements, it seems like this PR is meant to have a followup PR, and the stdout printing is a best approximation of debugging for development purposes.  If that is the case, then instead of removing the stdout printing prematurely, add a few TODO comments to remove the stdout printing.

Reply:
Update code, fixing many lint complaints.